### PR TITLE
DOV-6118: Config-Rewrite of the Push Notification Plugin

### PR DESCRIPTION
**Pull request by fleischie**
Rewrite the documentation of the currently implemented push notification plugins.

Note: There are http-client settings that can be used in the plugins, that are currently not documented, yet. This will be done in another PR.

### CODE REVIEW

File: source/configuration_manual/push_notification.rst [hunk 1]
```diff
@@ -72,43 +72,53 @@ Drivers
...
+notifications into the debug log of the process.
+
+Configuration options:
+
+===================================== ========== ================ ==========================================================================
+Name                                  Required   Type             Description
+===================================== ========== ================ ==========================================================================
+``driver``                            **YES**    :ref:`string`    To identify this settings block the driver should get the value ``dlog``.
+
+``push_notification_dlog_enabled``    NO         :ref:`boolean`   Whether to enabled this plugin or not. (DEFAULT: ``no``)
```
Comment (sirainen) [inline source/configuration_manual/push_notification.rst:112]: This was removed

File: source/configuration_manual/push_notification.rst [hunk 2]
```diff
@@ -72,43 +72,53 @@ Drivers
...
+===================================== ========== ================ ==========================================================================
+
 .. code-block:: none
 
-  plugin {
-    push_notification_driver = dlog
+  push_notification dlog {
+    driver = dlog
+    push_notification_dlog_enabled = yes
```
Comment (sirainen) [inline source/configuration_manual/push_notification.rst:119]: here also

File: source/configuration_manual/push_notification.rst [hunk 3]
```diff
@@ -123,34 +133,35 @@ used by any push endpoint that implements this API, not just OX App Suite.
...
 
 .. code-block:: none
 
-  plugin {
-    push_notification_driver = ox:url=http://login:pass@node1.domain.tld:8009/preliminary/http-notify/v1/notify user_from_metadata timeout_msecs=10000
+  push_notification ox {
+    driver = ox
+    push_notification_ox_url = http://login:pass@node1.domain.tld:8009/preliminary/http-notify/v1/notify
+    push_notification_ox_user_from_metadata = yes
+    push_notification_ox_cache_ttl = 10secs
```
Comment (sirainen) [inline source/configuration_manual/push_notification.rst:164]: These could be also simply `ox_url`, `ox_user_from_metadata`. Not quite sure whether that's nicer or not. Maybe more confusing..

**Review by sirainen — CHANGES_REQUESTED** [review]

File: source/configuration_manual/push_notification.rst (hunk 3 above)
Comment (fleischie) [reply source/configuration_manual/push_notification.rst:162]: I opt for the long and explicit version. This makes it much more clear, without raising any further questions.

I _could_ add a note regarding "The `push_notification_` prefix is optional but suggested. This document will use the verbose version."

**Review by cmouse — APPROVED** [review]

**Pull request merged**